README.md

```diff
@@ -65,8 +65,21 @@ Point your browser to: [http://localhost:8000/pages/blockly.html](http://localho
 - Design you own block and then add the metadata to: `blocks` and `generators` in the block repository. An xml snippet should be included in any Javascript file of the block repository, please see [sr_blockly](https://github.com/shadow-robot/sr_blockly/tree/master/sr_blockly_blocks/toolbox) for an example of how to do this.
 - catkin_make will automatically merge all Javascript files into a single file. So in case if you have changed Javascript file you need to run catkin_make.
 
+### Advanced
+
+#### Saving functions
+Functions can be created with the standard in-built function blocks. If you wish to save a function as a new block, select 'Save Function' from the Builder menu. 
+You may select in which category you wish to save the new block, this can be a new category that is specified by appending '/' and the name of the new category to the existing category name, for example:
+
+Select existing category: Variables\
+Or to place block in a new category: Variables/new_category_name
+
+The 'Function' dropdown allows you to select the function that you wish to save.
+Once 'Save' has been selected, a Javascript file will be downloaded. This file should be added to your block repository toolbox directory where the existing category Javascript files reside.
+catkin_make should then be run to rebuild blocks_uncompressed.js.
```

Comment:
maybe you should also mention that for now it might require full rebuild. Because new files are not detected.